gh-108202: `calendar`: Document `TextCalendar` methods (#127608)
python/cpython: Doc/library/calendar.rst

   :class:`TextCalendar` instances have the following methods:

   .. method:: formatday(theday, weekday, width)

>       Return a string representing a single day formatted with the given width.

Reviewer: ```suggestion
      Return a string representing a single day formatted with the given *width*.
```

   :class:`TextCalendar` instances have the following methods:

>    .. method:: formatday(theday, weekday, width=2)

Reviewer 2: The `width=2` suggested to me that one can pass this a keyword argument, but that is not the case. Also there does not seem to be a default value for the `TextCalendar` and the `weekday` argument is not used at all. 

https://github.com/python/cpython/blob/b9a492b809d8765ee365a5dd3c6ba4e5130a80af/Lib/calendar.py#L342-L350

I tried looking at the unit tests to see what the method actually does, but I cannot find any.

      Return a :class:`str` representing a single day formatted with the given width.
      If *theday* is ``0``, return a :class:`str` of spaces of

>       the specified width, representing an empty day. The *weekday* parameter
>       represents the day of the week, where ``0`` is Monday and ``6`` is Sunday.

Reviewer 2: ```suggestion
      the specified width, representing an empty day. The *weekday* parameter
      is unused.
```

>    .. method:: formatweek(theweek, width=2)

Reviewer 2: ```suggestion
   .. method:: formatweek(theweek, width)
```

      Return a :class:`str` representing an entire week formatted with the given width for each day.
      The *theweek* parameter is a list of tuples, where each tuple contains a day of
      the month or ``0`` for padding, and the corresponding weekday where ``0`` is
      Monday and ``6`` is Sunday. Each day is padded to the specified width.

>    .. method:: formatweekday(weekday, width=2)

Reviewer 2: ```suggestion
   .. method:: formatweekday(weekday, width)
```

      Return a :class:`str` representing the name of a single weekday formatted to
      the specified width. The *weekday* parameter is an integer representing
      the day of the week, where ``0`` is Monday and ``6`` is Sunday.

>    .. method:: formatweekheader(width=2)

Reviewer 2: ```suggestion
   .. method:: formatweekheader(width)
```

   :class:`TextCalendar` instances have the following methods:

   .. method:: formatday(theday, weekday, width)

>       Return a :class:`str` representing a single day formatted with the given width.

Reviewer: We generally say "string" without saying the exact type because it renders a bit better in a text (for instance, we don't markup "list" when we talk about a "list of [...]").

      If *theday* is ``0``, return a :class:`str` of spaces of
      the specified width, representing an empty day. The *weekday* parameter
      is unused.

   .. method:: formatweek(theweek, w=0)

      Return a single week in a string with no newline. If *w* is provided, it
      specifies the width of the date columns, which are centered. Depends
      on the first weekday as specified in the constructor or set by the
      :meth:`setfirstweekday` method.

   .. method:: formatweekday(weekday, width)

      Return a :class:`str` representing the name of a single weekday formatted to

>       the specified width. The *weekday* parameter is an integer representing

Reviewer: ```suggestion
      the specified *width*. The *weekday* parameter is an integer representing
```

      the day of the week, where ``0`` is Monday and ``6`` is Sunday.

   .. method:: formatweekheader(width)

      Return a :class:`str` containing the header row of weekday names, formatted
      with the given width for each column. The names depend on the locale

>       settings and are padded to the specified width.

Reviewer: ```suggestion
      settings and are padded to the specified *width*.
```

   .. method:: formatmonth(theyear, themonth, w=0, l=0)

      Return a month's calendar in a multi-line :class:`str`. If *w* is provided, it
      specifies the width of the date columns, which are centered. If *l* is
      given, it specifies the number of lines that each week will use. Depends
      on the first weekday as specified in the constructor or set by the
      :meth:`setfirstweekday` method.

   .. method:: formatmonthname(theyear, themonth, width=0, withyear=True)

      Return a :class:`str` representing the month's name centered within the
      specified width. If *withyear* is ``True``, include the year in the

>       output. The *theyear* and *themonth* parameters specify the year
>       and month for the name to be formatted.

Reviewer: ```suggestion
      output. The *theyear* and *themonth* parameters specify the year
      and month for the name to be formatted respectively.
```

   :class:`TextCalendar` instances have the following methods:

   .. method:: formatday(theday, weekday, width=2)

>       Return a string representing a single day formatted with the given width

Reviewer 3: Typo? Also, a comma is probably needed here.
```suggestion
      Return a string representing of a single day formatted with the given width,
```

Author: I don't believe "of" is necessary and neither is the comma.

>       which defaults to 2. If *theday* is ``0``, return a string of spaces of

Reviewer 3: Let's be consistent with putting numbers in a code block or not.
```suggestion
      which defaults to ``2``. If *theday* is ``0``, return a string of spaces of
```

      the specified width, representing an empty day. The *weekday* parameter
      represents the day of the week, where ``0`` is Monday and ``6`` is Sunday.

   .. method:: formatweek(theweek, width=2)

      Return a string representing an entire week formatted with the given width for each day.
      The *theweek* parameter is a list of tuples, where each tuple contains a day of

>       the month or ``0`` for padding, and the corresponding weekday where 0 is
>       Monday and 6 is Sunday. Each day is padded to the specified width.

Reviewer 3: ```suggestion
      the month or ``0`` for padding, and the corresponding weekday where ``0`` is
      Monday and ``6`` is Sunday. Each day is padded to the specified width.
```

   .. method:: formatweekday(weekday, width=2)

      Return a string representing the name of a single weekday formatted to
      the specified width. The *weekday* parameter is an integer representing

>       the day of the week, where 0 is Monday and 6 is Sunday.

Reviewer 3: ```suggestion
      the day of the week, where ``0`` is Monday and ``6`` is Sunday.
```

      on the first weekday as specified in the constructor or set by the
      :meth:`setfirstweekday` method.

   .. method:: formatmonthname(theyear, themonth, width=0, withyear=True)

      Return a string representing the month's name centered within the

>       specified width. If *withyear* is True, include the year in the

Reviewer 3: Constants need code blocks:
```suggestion
      specified width. If *withyear* is ``True``, include the year in the
```

.. function:: month(theyear, themonth, w=0, l=0)

>    Returns a month's calendar in a multi-line :class:`str` using the :meth:`~TextCalendar.formatmonth`

Reviewer 2: The original formulation was also fine with me. In general we try to keep code changes minimal, so reverting this is also an option. (similar remark for the other changes like this)

Reviewer: I agree on reverting this.

   A sequence that represents the months of the year in the current locale.  This
   follows normal convention of January being month number 1, so it has a length of

>    13 and  ``month_name[0]`` is the empty :class:`str`.

Reviewer 3: We only want one space before the code block, right?
```suggestion
   13 and ``month_name[0]`` is the empty :class:`str`.
```
